java/sage/epg/sd/SDRecommended.java

```diff
@@ -0,0 +1,470 @@
+          {
+            url = seriesInfo.getImageURL(0, true);
+          }
+        } catch (NumberFormatException e) {}
+      }
+    }
+
+    return url == null ? "" : url;
+  }
+
+  /**
+   * Reduce the number of editorial candidates to a specific number by occurrence.
+   * <p/>
+   * This will also retrieve program details and add them to the objects because it's more efficient
+   * to do it this way since we are already iterating through the editorials.
+   */
+  public static void reduceByWeight(List<SDEditorial> editorials, int limit) throws IOException, SDException
+  {
+    Collections.sort(editorials, SDEditorial.WEIGHT_COMPARATOR);
+    limit = Math.min(editorials.size(), limit);
+    List<String> query = new ArrayList<>(limit);
+    for (int i = 0; i < editorials.size(); i++)
+    {
+      SDEditorial editorial = editorials.get(i);
+      String programId = editorial.getAiring().getShow().getExternalID();
+
+      // If we are already over the limit, try to limit the imports to things we actually have
+      // artwork to display. This also helps ensures that the show to be displayed is one that
+      // someone cared enough to provide artwork and that often also means that the show is
+      // fairly well received.
+      if (editorials.size() > limit)
+      {
+        SDProgram program = editorial.getProgram();
+        if (program != null && !program.hasImageArtwork())
+        {
+          editorials.remove(i--);
+          continue;
+        }
+      }
+
+      query.add(programId);
+      if (query.size() >= limit)
+      {
+        while (i < editorials.size())
+        {
+          editorials.remove(i);
+        }
+        break;
+      }
+    }
+
+    SDProgram programs[] = SDRipper.ensureSession().getPrograms(query);
+    for (int i = 0; i < programs.length; i++)
+    {
+      SDEditorial editorial = editorials.get(i);
+      editorial.setProgram(programs[i]);
+    }
+
+    if (Sage.DBG) System.out.println("SDEPG Recommended: " + editorials.toString());
+  }
+
+  public static List<SDEditorial> getUsable(List<SDEditorial> editorials)
+  {
+    int alreadyExists = 0;
+    List<SDEditorial> returnValues = new ArrayList<>();
+    Agent favorites[] = Wizard.getInstance().getFavorites();
+    TVEditorial currentEditorials[] = Wizard.getInstance().getEditorials();
+    for (SDEditorial editorial : editorials)
+    {
+      SDRecommendation recommendation = editorial.getRecommendation();
+      String title = recommendation.getTitle120();
+      if (title == null || title.length() == 0)
+        continue;
+      String programId = recommendation.getProgramId();
+      if (programId == null || programId.length() == 0)
+        continue;
+
+      // Don't recommend movies that we have already seen or don't exist in the Wizard.
+      if (programId.startsWith("MV"))
+      {
+        Show show;
+        if ((show = Wizard.getInstance().getShowForExternalID(programId)) == null || show.isWatched())
```

Comment:
This looks like it will skip anything with an ID that starts with 00 because this first lookup on the uncorrected program ID will always fail in that case.